### PR TITLE
[android] Fix crash if decode .heic image in 32bit mode android app.

### PR DESCRIPTION
**Pull request by toneyzeng**
The type |long| is equal to int32_t in 32bit app, but int64_t in 64bit app. 
The type |jlong| is always equal to int64_t.
CallStaticObjectMethod uses "J" to request a jlong type which needs 8 byte data but given |long| value which provides 4 byte data in 32bit mode will trigger java value error and at last a wrong pointer to AndroidImageGenerator object.

### CODE REVIEW

**Comment by chinmaygarde** [comment]
cc @jason

File: shell/platform/android/android_image_generator.cc [hunk 1]
```diff
@@ -106,7 +115,7 @@ void AndroidImageGenerator::DoDecodeImage() {
   auto bitmap = std::make_unique<fml::jni::ScopedJavaGlobalRef<jobject>>(
       env, env->CallStaticObjectMethod(g_flutter_jni_class->obj(),
                                        g_decode_image_method, direct_buffer,
-                                       reinterpret_cast<long>(this)));
+                                       Pointer2Jlong(this)));
```
Comment (jason-simmons) [inline shell/platform/android/android_image_generator.cc:118]: Use `reinterpret_cast<jlong>` here

File: shell/platform/android/android_image_generator.cc [hunk 2]
```diff
@@ -191,7 +200,7 @@ void AndroidImageGenerator::NativeImageHeaderCallback(JNIEnv* env,
                                                       int width,
                                                       int height) {
   AndroidImageGenerator* generator =
-      reinterpret_cast<AndroidImageGenerator*>(generator_address);
+      Jlong2Pointer<AndroidImageGenerator*>(generator_address);
```
Comment (jason-simmons) [inline shell/platform/android/android_image_generator.cc:203]: This change is not needed.  The original `reinterpret_cast<AndroidImageGenerator*>` should be safe here.

File: shell/platform/android/android_image_generator.cc (hunk 1 above)
Comment (toneyzeng) [reply shell/platform/android/android_image_generator.cc]: OK. It looks well.

File: shell/platform/android/android_image_generator.cc (hunk 2 above)
Comment (toneyzeng) [reply shell/platform/android/android_image_generator.cc]: Done.

**Review by jason-simmons — APPROVED** [review]

**Pull request merged**